java/layer-wrapper/scripts/otel-proxy-handler

```diff
@@ -9,4 +9,6 @@ fi
 
 export OTEL_INSTRUMENTATION_AWS_LAMBDA_FLUSH_TIMEOUT=10000
 
-exec "$@"
+# java17 runtime puts the handler to run as a command line argument and seems to prefer
+# this over _HANDLER, so apply a regex to replace the original handler name with our wrapper
+exec "${@//$OTEL_INSTRUMENTATION_AWS_LAMBDA_HANDLER/$_HANDLER}"
```

Comment:
@johnbley did you confirm this doesn't break anything with Java 11?  I assume it doesn't but just want to be sure.
(I think it would still help to include an example string before/after in the PR description on both Java 11 and Java 17 -- unless they're the same.)

Reply:
java17: 
```
sun.java.command=com.amazonaws.services.lambda.runtime.api.client.AWSLambda example.Hello::handleRequest
```

java11: 
```
sun.java.command=lambdainternal.AWSLambda
```

And, yes, I verified with our internal tests that this regex works for both 11 and 17.

Reply:
For the java17 case you can see in their code that they clearly ignore `_HANDLER` and expect a single argument to be that handler string: https://github.com/aws/aws-lambda-java-libs/blob/90155b3db0f59f93d442eb533e735e9e0041b552/aws-lambda-java-runtime-interface-client/src/main/java/com/amazonaws/services/lambda/runtime/api/client/AWSLambda.java#L187

Reply:
Thanks!